apps/shipments/views/shipment.py

```diff
@@ -178,6 +179,38 @@ def tracking(self, request, version, pk):
 
         raise NotFound("No tracking data found for Shipment.")
 
+    @method_decorator(cache_page(60 * 60, remember_all_urls=True))  # Cache responses for 1 hour
+    @action(detail=True, methods=['get'], permission_classes=(IsOwnerOrShared,))
+    def telemetry(self, request, version, pk):
+        """
+        Retrieve tracking data from db
+        """
+        LOG.debug(f'Retrieve telemetry data for a shipment {pk}.')
+        log_metric('transmission.info', tags={'method': 'shipments.tracking', 'module': __name__})
+        shipment = Shipment.objects.get(pk=pk)
+
+        sensor_id = request.query_params.get('sensor_id', None)
+        hardware_id = request.query_params.get('hardware_id', None)
+
+        # TODO: implement device/shipment authorization for telemetry data
+
+        telemetry_data = TelemetryData.objects.filter(shipment__id=shipment.id)
+        if sensor_id:
+            telemetry_data = telemetry_data.filter(sensor_id=sensor_id)
+
+        if hardware_id:
+            telemetry_data = telemetry_data.filter(hardware_id=hardware_id)
+
+        if telemetry_data.exists():
+            response = Template('{"data": $geojson}').substitute(
+                geojson=render_filtered_telemetry(shipment, telemetry_data)
```

Comment:
Are you not serializing this data as GeoJSON? Because GeoJSON is referenced multiple times in this view, as well as in the render_filtered_telemtry function.

I don't think telemetry data is associated with tracking data, so I think you can just render the telemetry data normally using the existing JSON API renderer, without the need for this function, or the Template().substitute optimization that was just used for rendering GeoJSON quickly.

Reply:
I am not no, the reason I had this was so that I could add force the telemetry data to return all the data at once, not just a set of 10 like django normally does.